pkg/cloudprovider/identity.go

```diff
@@ -65,7 +66,7 @@ func filter(ls *[]string, filter string) {
 	}
 
 	for i, v := range *ls {
-		if v == filter {
+		if strings.EqualFold(v, filter) {
```

Comment:
One thing to note - before the identity was never removed if it was manually created and the resourceID format was different in azure identity. With this change it'll be deleted even if the case is different in resourceID.